changelog/unreleased/pull-2773

```diff
@@ -0,0 +1,8 @@
+Enhancement: Optimize repository implementation
+
+We've optimized the implementation of handling index entries in the repository.
+
+Restic now uses less memory for backups which add a lot of data, e.g. large initial backups.
+Also it is more stable in some edge situations and allows easier future enhancments.
```

Comment:
After line 3 is removed, the changelog no longer refers to the implementation such that this should be changed to
"In addition, we've improved the stability in some edge cases." .

Reply:
changed it.